Assets/Scripts/Player/Input Parser/InputParser.cs

```diff
@@ -27,13 +32,31 @@ private void OnDisable()
         private void AddListeners()
         {
             _inputActionAsset["MoveTowards"].performed += MoveTowards;
+            _inputActionAsset["Interact"].performed += Interact;
         }
 
         private void RemoveListeners()
         {
             _inputActionAsset["MoveTowards"].performed -= MoveTowards;
+            _inputActionAsset["Interact"].performed -= Interact;
         }
 
         private void MoveTowards(InputAction.CallbackContext context) => _playerMovement.StartMovingEvent();
+
+        private void Interact(InputAction.CallbackContext context)
+        {
+            Ray ray = _mainCamera.ScreenPointToRay(SetMousePos());
+            Physics.Raycast(ray.origin, ray.direction *10, out RaycastHit hit);
+            
+            if (hit.collider
+                && hit.collider.CompareTag(INTERACTABLE_TAG))
+                hit.collider.transform.GetComponent<InteractableObject>().onInteract.Invoke();
+        }
+
+        private Vector3 SetMousePos()
+        {
+            Vector2 mousePos = _inputActionAsset["MousePosition"].ReadValue<Vector2>();
+            return mousePos;
+        }
```

Comment:
```suggestion
private Vector3 GetMousePosition() => (Vector3) _input["MousePosition"].ReadValue<Vector2>();
```
Dit kan op 1 line en functie rename.